### PR TITLE
Add more waiting functions

### PR DESCRIPTION
**Pull request by Ntr0**
Add *AndWait for 
* Create/Update/Delete Datacenter/Server/Lan

### CODE REVIEW

File: datacenter.go [hunk 1]
```diff
@@ -79,6 +80,19 @@ func (c *Client) CreateDatacenter(dc Datacenter) (*Datacenter, error) {
...
 
+// CreateDatacenterAndWait creates a data center, waits for the request to finish and returns a refreshed
+// result.
+func (c *Client) CreateDatacenterAndWait(ctx context.Context, dc Datacenter) (*Datacenter, error) {
+	rsp, err := c.CreateDatacenter(dc)
+	if err != nil {
+		return nil, err
+	}
+	if err := c.WaitTillProvisionedOrCanceled(ctx, rsp.Headers.Get("location")); err != nil {
+		return nil, err
```
Comment (avorima) [inline datacenter.go]: You should return the created objects when wait fails to at least preserve the resource id

File: datacenter.go [hunk 2]
```diff
@@ -95,9 +109,30 @@ func (c *Client) UpdateDataCenter(dcid string, obj DatacenterProperties) (*Datac
...
 }
 
+// UpdateDatacenter updates a data center, waits for the request to finish and returns a refreshed result.
+func (c *Client) UpdateDatacenterAndWait(ctx context.Context, dcid string, obj DatacenterProperties) (*Datacenter, error) {
+	rsp, err := c.UpdateDataCenter(dcid, obj)
+	if err != nil {
+		return nil, err
+	}
+	if err := c.WaitTillProvisionedOrCanceled(ctx, rsp.Headers.Get("location")); err != nil {
+		return nil, err
```
Comment (avorima) [inline datacenter.go]: Same here, probably a good idea to return the object

File: server.go [hunk 3]
```diff
@@ -79,30 +92,52 @@ func (c *Client) GetServer(dcid, srvid string) (*Server, error) {
 }
 
 // UpdateServer partial update of server properties passed in as jason []byte
```
Comment (avorima) [inline server.go]: Please fix the comment while you're at it

File: datacenter.go (hunk 1 above)
Comment (Ntr0) [reply datacenter.go]: good point

File: datacenter.go [hunk 4]
```diff
@@ -82,15 +82,22 @@ func (c *Client) CreateDatacenter(dc Datacenter) (*Datacenter, error) {
...
+// the request or when refreshing the resource.
+func (c *Client) CreateDatacenterAndWait(ctx context.Context, dc Datacenter) (res *Datacenter, err error) {
+	res, err = c.CreateDatacenter(dc)
 	if err != nil {
-		return nil, err
+		return
 	}
-	if err := c.WaitTillProvisionedOrCanceled(ctx, rsp.Headers.Get("location")); err != nil {
-		return nil, err
+	if err := c.WaitTillProvisionedOrCanceled(ctx, res.Headers.Get("location")); err != nil {
```
Comment (avorima) [inline datacenter.go]: You're shadowing errors here and elsewhere

File: server.go [hunk 5]
```diff
@@ -72,13 +72,16 @@ func (c *Client) CreateServer(dcid string, server Server) (*Server, error) {
...
+	res, err = c.CreateServer(dcid, srvid)
 	if err != nil {
-		return nil, err
+		return
 	}
 	if err := c.WaitTillProvisionedOrCanceled(ctx, res.Headers.Get("location")); err != nil {
-		return nil, err
+		return
 	}
 	return c.GetServer(dcid, res.ID)
```
Comment (avorima) [inline server.go]: Missing error check

**Comment by avorima** [comment]
LGTM

**Comment by pennycoders** [comment]
Thanks for the contributions guys! Outstanding work keeping all of these PR's coming!

**Pull request merged**